.gitlab-ci.yml

```diff
@@ -45,7 +46,7 @@ lint:
   stage: linting_and_dependencies
   script:
     - pwd
-    - virtualenv -p python3.6 venv && . venv/bin/activate && make prepare && make develop extras=[all] packages=htcondor
+    - virtualenv -p ${MAIN_PYTHON_PKG} venv && . venv/bin/activate && make prepare && make develop extras=[all] packages=htcondor
```

Comment:
These days I do `python3.8 -m venv` or similar instead of virtualenv

Reply:
@mr-c That would be fine for testing where the base python is guaranteed to be clean, but our `make` commands enforce a virtualenv for users, where I think it's a good way of avoiding some of the more common problems where users may have done God knows what to their default python packages.  So I'd like to keep this for now.